### PR TITLE
return diff information in file module touch operation in check_mode

### LINKED ISSUE #50452: File module should show return a changed=true with check_mode=true and state=touch
##### SUMMARY
Currently the file module does absolutly nothing if it is called with state=touch and check_mode=true. It should return at least a changed=true state.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

lib/ansible/modules/files/file.py

##### ANSIBLE VERSION

```
ansible 2.7.5
```

##### ADDITIONAL INFORMATION
According to this line https://github.com/ansible/ansible/blob/v2.7.5/lib/ansible/modules/files/file.py#L448 there is no code that handle the check_mode.
I had noticed some github's issues related to bad behaviour with state=touch and missing directories... but I can cite one of your developper by saying "check mode should show all changes without doing any real change"
As I can see, the touch was always triggering a changed=true state until this commit https://github.com/ansible/ansible/commit/a78cc15099d6a31e948eb5aa6f9f3cc82337f5bc that remove the {changed: true} attribute from "result" variable.
So could you please consider add a "else" case with result['changed'] = True
To show up at least that the task will perform a change.

```yaml
- name: Ensure file for specifications
      file:
        path: /tmp/test
        state: touch
```

With --check

```
TASK [Ensure file for specifications] *****************************************************************************************************************************************************************************************
ok: [host] => {"changed": false, "path": "/tmp/test", "state": "absent"}
```

Without --check

```
TASK [Ensure file for specifications] ***************************************************************************************************************************************************************************************
--- before
+++ after
@@ -1,6 +1,6 @@
 {
-    "atime": 1546440986.1151996,
-    "mtime": 1546440986.1151996,
+    "atime": 1546440986.119353,
+    "mtime": 1546440986.119352,
     "path": "/tmp/test",
-    "state": "absent"
+    "state": "touch"
 }

```

### PR DESCRIPTION
**Pull request by Turgon37**
##### SUMMARY
Add diff and result informations on file module with state "touch" even in check mode

fixes #50452

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/files/file.py

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by samdoran** [comment]
Please create a changelog fragment. See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

File: lib/ansible/modules/files/file.py [hunk 1]
```diff
@@ -530,6 +530,11 @@ def execute_touch(path, follow, timestamps):
 
         result['changed'] = changed
         result['diff'] = diff
+    else:
+        diff = initial_diff(path, 'touch', prev_state)
+        # Even if check_mode = True, touch always changes the file because it updates its timestamp
+        result['changed'] = True
```
Comment (s-hertel) [inline lib/ansible/modules/files/file.py:536]: If the `access_time` and `modification_time` options are set to "preserve", check mode is still going to show changed=True, even though no change will be made without check mode.

**Review by s-hertel — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@Turgon37 Are you still working on this? Thanks.

needs_info

**Comment by Turgon37** [comment]
Hello yes It's in my todo list :)

**Comment by mattclay** [comment]
Thanks for your interest in working on this. Since this PR hasn't been updated for some time it is being closed. If you're interested in continuing work on this, please open a new PR.

**Pull request closed without merging**